diff --git a/Makefile b/Makefile
index 2d1bda1..596d5b4 100644
--- a/Makefile
+++ b/Makefile
@@ -8,26 +8,28 @@ OS ?= LINUX
 # work.
 #USE_LIBUSB ?= YES
 
+OUTDIR ?= build
+
 ifeq ($(OS), LINUX)  # also works on FreeBSD
 CC ?= gcc
 CFLAGS ?= -O2 -Wall
-teensy_loader_cli: teensy_loader_cli.c
-	$(CC) $(CFLAGS) $(CPPFLAGS) -s -DUSE_LIBUSB -o teensy_loader_cli teensy_loader_cli.c -lusb $(LDFLAGS)
+$(OUTDIR)/teensy_loader_cli: teensy_loader_cli.c
+	$(CC) $(CFLAGS) $(CPPFLAGS) -s -DUSE_LIBUSB -o $(OUTDIR)/teensy_loader_cli teensy_loader_cli.c -lusb-1.0 $(LDFLAGS)
 
 
 else ifeq ($(OS), WINDOWS)
 CC ?= i586-mingw32msvc-gcc
 CFLAGS ?= -O2 -Wall
-teensy_loader_cli.exe: teensy_loader_cli.c
-	$(CC) $(CFLAGS) -s -DUSE_WIN32 -o teensy_loader_cli.exe teensy_loader_cli.c -lhid -lsetupapi -lwinmm
+$(OUTDIR)/teensy_loader_cli.exe: teensy_loader_cli.c
+	$(CC) $(CFLAGS) -s -DUSE_LIBUSB -o $(OUTDIR)/teensy_loader_cli.exe teensy_loader_cli.c -lusb-1.0 $(LDFLAGS)
 
 
 else ifeq ($(OS), MACOSX)
 ifeq ($(USE_LIBUSB), YES)
 CC ?= gcc
 CFLAGS ?= -O2 -Wall
-teensy_loader_cli: teensy_loader_cli.c
-	$(CC) $(CFLAGS) -s -DUSE_LIBUSB -DMACOSX -o teensy_loader_cli teensy_loader_cli.c -lusb -I /usr/local/include -L/usr/local/lib
+$(OUTDIR)/teensy_loader_cli: teensy_loader_cli.c
+	$(CC) $(CFLAGS) -DUSE_LIBUSB -DMACOSX -o $(OUTDIR)/teensy_loader_cli teensy_loader_cli.c -lusb $(LDFLAGS)
 	 
 else
 CC ?= gcc
@@ -37,19 +39,19 @@ SDK ?= $(shell xcrun --show-sdk-path)
 #CC = /Developer_xcode32/usr/bin/gcc-4.0
 #CFLAGS = -O2 -Wall -arch i386 -arch ppc
 CFLAGS ?= -O2 -Wall
-teensy_loader_cli: teensy_loader_cli.c
+$(OUTDIR)/teensy_loader_cli: teensy_loader_cli.c
 ifeq ($(SDK),)
 	$(error SDK was not found. To use this type of compilation please install Xcode)
 endif
-	$(CC) $(CFLAGS) -DUSE_APPLE_IOKIT -isysroot $(SDK) -o teensy_loader_cli teensy_loader_cli.c -Wl,-syslibroot,$(SDK) -framework IOKit -framework CoreFoundation
+	$(CC) $(CFLAGS) -DUSE_APPLE_IOKIT -isysroot $(SDK) -o $(OUTDIR)/teensy_loader_cli teensy_loader_cli.c -Wl,-syslibroot,$(SDK) -framework IOKit -framework CoreFoundation
 
 endif
 
 else ifeq ($(OS), BSD)  # works on NetBSD and OpenBSD
 CC ?= gcc
 CFLAGS ?= -O2 -Wall
-teensy_loader_cli: teensy_loader_cli.c
-	$(CC) $(CFLAGS) -s -DUSE_UHID -o teensy_loader_cli teensy_loader_cli.c
+$(OUTDIR)/teensy_loader_cli: teensy_loader_cli.c
+	$(CC) $(CFLAGS) -s -DUSE_UHID -o $(OUTDIR)/teensy_loader_cli teensy_loader_cli.c
 
 
 endif
diff --git a/teensy_loader_cli.c b/teensy_loader_cli.c
index 523f2af..8904074 100644
--- a/teensy_loader_cli.c
+++ b/teensy_loader_cli.c
@@ -1029,6 +1032,9 @@ int printf_verbose(const char *format, ...)
 	return 0;
 }
 
+#ifdef WIN32
+#include <windows.h>
+#endif
 void delay(double seconds)
 {
 	#ifdef WIN32
